deploy/AllocatorInit.sol

```diff
@@ -0,0 +1,278 @@
+
+        // Allow the automation contracts to operate on the funnels
+        RolesLike(sharedInstance.roles).setUserRole(ilk, networkInstance.stableSwapper,        cfg.automationRole, true);
+        RolesLike(sharedInstance.roles).setUserRole(ilk, networkInstance.stableDepositorUniV3, cfg.automationRole, true);
+
+        RolesLike(sharedInstance.roles).setRoleAction(ilk, cfg.automationRole, networkInstance.swapper,        SwapperLike.swap.selector,            true);
+        RolesLike(sharedInstance.roles).setRoleAction(ilk, cfg.automationRole, networkInstance.depositorUniV3, DepositorUniV3Like.deposit.selector,  true);
+        RolesLike(sharedInstance.roles).setRoleAction(ilk, cfg.automationRole, networkInstance.depositorUniV3, DepositorUniV3Like.withdraw.selector, true);
+        RolesLike(sharedInstance.roles).setRoleAction(ilk, cfg.automationRole, networkInstance.depositorUniV3, DepositorUniV3Like.collect.selector,  true);
+
+        // Set the allocator proxy as the ilk admin instead of the Pause Proxy
+        RolesLike(sharedInstance.roles).setIlkAdmin(ilk, cfg.allocatorProxy);
+
+        // Allow facilitator to set configurations in the automation contracts
+        WardsLike(networkInstance.stableSwapper).rely(cfg.facilitator);
+        WardsLike(networkInstance.stableDepositorUniV3).rely(cfg.facilitator);
+        WardsLike(networkInstance.conduitMover).rely(cfg.facilitator);
+
+        // Add keepers to the automation contracts
+        KissLike(networkInstance.stableSwapper).kiss(cfg.stableSwapperKeeper);
+        KissLike(networkInstance.stableDepositorUniV3).kiss(cfg.stableDepositorUniV3Keeper);
+        KissLike(networkInstance.conduitMover).kiss(cfg.conduitMoverKeeper);
+
+        // Move ownership of the network contracts to the allocator proxy
+        switchOwner(networkInstance.vault,                networkInstance.owner, cfg.allocatorProxy);
+        switchOwner(networkInstance.buffer,               networkInstance.owner, cfg.allocatorProxy);
+        switchOwner(networkInstance.swapper,              networkInstance.owner, cfg.allocatorProxy);
+        switchOwner(networkInstance.depositorUniV3,       networkInstance.owner, cfg.allocatorProxy);
+        switchOwner(networkInstance.stableSwapper,        networkInstance.owner, cfg.allocatorProxy);
+        switchOwner(networkInstance.stableDepositorUniV3, networkInstance.owner, cfg.allocatorProxy);
+        switchOwner(networkInstance.conduitMover,         networkInstance.owner, cfg.allocatorProxy);
+
+        // Add allocator-specific contracts to changelog
+        dss.chainlog.setAddress(cfg.vaultClKey,  networkInstance.vault);
+        dss.chainlog.setAddress(cfg.bufferClKey, networkInstance.buffer);
+
+        // Add to ilk registry
+        IlkRegistryLike(cfg.ilkRegistry).put({
+            _ilk    : ilk,
+            _join   : address(0),
+            _gem    : address(0),
+            _dec    : 0,
+            _class  : 3,
```

Comment:
Weren't we using 3 for something else?
We probably want a fresh number here.

Reply:
Ah yes, I miss-remembered.
So rwas are 3.
Both d3ms and teleport are 4.
Just verified there is nothing of class 5 yet so using that.
Done.